Library/Homebrew/search.rb

```diff
@@ -22,19 +22,29 @@ def self.search_descriptions(string_or_regex, args, search_type: :desc)
 
       if args.formula? || both
         ohai "Formulae"
-        CacheStoreDatabase.use(:descriptions) do |db|
-          cache_store = DescriptionCacheStore.new(db)
-          Descriptions.search(string_or_regex, search_type, cache_store, eval_all).print
+        if eval_all
+          CacheStoreDatabase.use(:descriptions) do |db|
+            cache_store = DescriptionCacheStore.new(db)
+            Descriptions.search(string_or_regex, search_type, cache_store, eval_all).print
+          end
+        else
+          descriptions = Homebrew::API::Formula.all_formulae.transform_values { |data| data["desc"] }
```

Comment:
I wonder if this warrants printing a warning if there's non-core taps with formulae tapped (as they won't be checked)?

Reply:
I think so. Was expecting to add this, just wanted to get the core logic committed first. Current code has warnings for both formulae and casks, and they only print if there are 3rd party taps installed that include formulae/casks.